lib/utils/conf_gen.py

```diff
@@ -457,10 +418,52 @@ async def __call__(self, context_dict: dict[str, Any]):
             self._helper(config_dict)
 
 
-class ConvertIndexedToList(ConfigPipelineStep):
-    def __init__(self):
-        super().__init__()
+class ConvertIndexesSubcontextsToNamedSection(ConfigPipelineStep):
+    """Converts subcontext sections that are indexed to named sections. E.g.:
+    "network":
+        {
+            "tls[0]": {
+               "name": "tls-name"
+            }
+        }
+    "network":
+        {
+            NamedSection("tls", "tls-name"): {
+               . . .
+            }
+        }
+    should be
+    """
+
+    def _helper(self, config_dict: dict[str | IntermediateKey, Any]):
+        for config in list(config_dict.keys()):
+            value = config_dict[config]
+
+            if isinstance(value, dict):
+                self._helper(config_dict[config])
+
+                if isinstance(config, InterUnnamedSectionKey):
+                    config_split = config.type.split("[")
 
+                    if len(config_split) > 1:
+                        if "name" in config_dict[config]:
+                            name = config_dict[config]["name"]
+                            del config_dict[config]["name"]
+
+                            del config_dict[config]
```

Comment:
```suggestion
                            del config_dict[config]
```
I think only 1 delete is needed

Reply:
It is needed. Ill rewrite it so it makes more sense though